src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<Response<T>> {
+    const statusCode = context.switchToHttp().getResponse().statusCode;
+    const req = context.switchToHttp().getRequest();
+    const userDetail = req?.user || {};
```

Comment:
PrimathonAICodeReviewer

Consider adding error handling for potential null values in `req` object before accessing properties like `user`. This can prevent runtime errors.
```diff
+     const userDetail = req?.user || {};
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->